tutorons/tests/css/test_query.py

```diff
@@ -7,10 +7,97 @@
 import json
 from bs4 import BeautifulSoup
 from django.test import Client
-
+import tinycss
 
 logging.basicConfig(level=logging.INFO, format="%(message)s")
 
+class FetchExplanationsForMatthewsCSSSelectorTest(unittest.TestCase):
+    
+    def setUp(self):
+        self.client = Client()
+
+    def get_regions(self, document):
+        resp = self.client.post('/css/scan', data={'origin': 'www.test.com', 'document': document})
+        regions = json.loads(resp.content)
+        return regions
+
+    def test_single_css_selector(self):
+        
+        string = "<html> <body> <code> sel { background-color: lightblue; } </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 1)
+        r = regions[0]
+        self.assertEqual(r['node'], 'HTML > BODY:nth-of-type(1) > CODE:nth-of-type(1)')
+        self.assertEqual(r['start_index'], 1)
+        self.assertEqual(r['end_index'], 3)
+
+    def test_multiple_css_selectors(self):
+        
+        string = "<html> <body> <code> sel { background-color: lightblue; }  h1 {color: navy; margin-left: 20px;} </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 2)
+        r1 = regions[0]
+        self.assertEqual(r1['node'], 'HTML > BODY:nth-of-type(1) > CODE:nth-of-type(1)')
+        self.assertEqual(r1['start_index'], 1)
+        self.assertEqual(r1['end_index'], 3)
+
+        r2 = regions[1]
+        self.assertEqual(r2['node'], 'HTML > BODY:nth-of-type(1) > CODE:nth-of-type(1)')
+        self.assertEqual(r2['start_index'], 39)
+        self.assertEqual(r2['end_index'], 40)
+
+    def test_empty_declaration(self):
+
+        string = "<html> <body> <code> sel { }  </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 0)
+
+    def test_malformed_declaration(self):
+
+        string = "<html> <body> <code> sel : }  </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 0)
+
+        string = "<html> <body> <code> sel {  </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 0)
+
+    def test_missing_declaration(self):
+
+        string = "<html> <body> <code> sel </code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 0)
+
+    def test_java_class(self):
+
+        string = "<html> <body> <code>  public class JavaClass{ protected JavaClass(int x){} public void main(String[] args){ }}</code> </body> </html>"
+        regions = self.get_regions(string)
+
+        self.assertEqual(len(regions), 0)
+
+    def test_w33_schools(self):
```

Comment:
Actually, the test_w33_schools example was the one I was talking about!  I wouldn't delete it, just modify it so that you test a simpler, easier to read text.